File: Python/find_writer.py
"""You have a set of rows with names of famous writers encoded inside. Each row
is divided into 2 parts by pipe char (|). The first part has a writer's name.
The second part is a "key" to generate a name.

Your goal is to go through each number in the key (numbers are separated by
space) left-to-right. Each number represents a position in the 1st part of a
row. This way you collect a writer's name which you have to output.

Your program should accept as its first argument a path to a filename.
"""
import sys

for line in open(sys.argv[1], 'r'):
    if line.strip() != '':
        out = ''
        letters, indicies = tuple(line.split('|'))
        indicies = [index.strip() for index in indicies.split(' ') if index != '']
        for index in indicies:
            out += letters[int(index) - 1]
        print(out)

File: Perl/find_writer.pl
#!/usr/bin/perl
use feature qw(say);
open(INFILE, "<$ARGV[0]") or die("Cannot open file $ARGV[0] for reading: $!");

my $output;
while(my $line = <INFILE>) {
    next if($line =~ m/^s$/);
    $line =~ s/(^s|s*$)//g;
    $output = '';
    @results = split('|', $line);
    say(@results[0]);
    $letters = $results[0];
    $indicies = $results[1];
    # say($letters);
    # say($indicies);
    for my $index ($indicies){
        $output .= $letters[$index - 1];
    }
    # say($output);
}
close(INFILE);